File: Tests/Metrics/EditDistanceTests.fs
namespace Tests.Metrics

open System
open Stringier.Metrics
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type EditDistanceTests() =

    [<DataTestMethod>]
    [<DataRow(1, "ram", "rom")>]
    [<DataRow(2, "ram", "rob")>]
    member _.``Hamming Distance``(exp:int32, src:string, oth:string) =
        Assert.AreEqual(exp, hamming src oth)

    [<DataTestMethod>]
    [<DataRow(1, "ram", "rom")>]
    [<DataRow(2, "ram", "rob")>]
    [<DataRow(3, "ram", "random")>]
    [<DataRow(2, "flaw", "lawn")>]
    member _.``Levenshtein Distance``(exp:int32, src:string, oth:string) =
        Assert.AreEqual(exp, levenshtein src oth)

File: Tests/EditDistanceTests.cs
using System;
using Stringier;
using static Stringier.Metrics;
using Xunit;

namespace Tests {
	public class EditDistanceTests {
		[Theory]
		[InlineData("ram", "ram", 0)]
		[InlineData("ram", "rom", 1)]
		[InlineData("ram", "rob", 2)]
		[InlineData("ram", "rma", 2)]
		public void Hamming(String source, String other, Int32 expected) => Assert.Equal(expected, HammingDistance(source, other));

		[Theory]
		[InlineData("ram", "rom", 1)]
		[InlineData("\u0072\u0061\u006D", "\u0072\u0061\u0301\u006D", 1)]
		[InlineData("\u0072\u00E1\u006D", "\u0072\u0061\u0301\u006D", 2)]
		[InlineData("ram", "rob", 2)]
		[InlineData("ram", "random", 3)]
		[InlineData("flaw", "lawn", 2)]
		public void Levenshtein(String source, String other, Int32 expected) => Assert.Equal(expected, LevenshteinDistance(source, other));
	}
}
